src/rdf4cpp/rdf/datatypes/registry/Util.hpp

```diff
@@ -43,10 +91,10 @@ constexpr Acc tuple_type_fold_impl(Acc init, F &&f, std::index_sequence<Ixs...>)
  * @param f a unary operation to map the elements
  * @return the tuple generated by applying f to each element in tup
  */
-template<typename F, typename TupleLike>
-constexpr auto tuple_map(TupleLike &&tup, F &&f) {
-    constexpr size_t tup_size = std::tuple_size_v<std::remove_cvref_t<TupleLike>>;
-    return tuple_util_detail::tuple_map_impl(std::forward<TupleLike>(tup), std::forward<F>(f), std::make_index_sequence<tup_size>{});
+template<typename F, typename Tuple> requires AccessibleTuple<std::remove_cvref_t<Tuple>>
```

Comment:
`std::remove_csref_t` would still leave e.g. volatile. I would always use `std::decay_t`. Same below

Reply:
I don't think that is correct
see: https://en.cppreference.com/w/cpp/types/remove_cvref

Reply:
Right, v is for volatile. After reading through the docs again, I would prefer `std::remove_csref_t` over `std:decay_t`. Easier to forsee what we will get out of applying it.